doc/book/getting-started/database-and-models.md

```diff
@@ -152,14 +152,16 @@ class AlbumTable
             return;
         }
 
-        if (! $this->getAlbum($id)) {
+        try {
+            if ($this->getAlbum($id)) {
```

Comment:
The exceptional case should be handled first right?

Reply:
You mean

```php
        try {
            $this->getAlbum($id) {
        } catch (RuntimeException $e) {
            throw new RuntimeException(sprintf(
                'Cannot update album with identifier %d; does not exist',
                $id
            ));
        }
        $this->tableGateway->update($data, ['id' => $id]);`
```

to have more flat code structure without the if block?

Reply:
Or is it about the change in general? Because without my changes the "Cannot update" exception will never be thrown as it will quit during getAlbum() and its Exception with "Could not find row" will be received by the function calling saveAlbum() instead of the actually intended Exception.

Reply:
What he's indicating is that the update operation should not be wrapped in the try/catch block, just like you did in your previous comment. If it's in the try/catch, we may actually catch an exception from the update operation, which will be a very different error than the album not existing!

I'll make the change during merge.